### PR TITLE
Implement new APIs for VCS Runbooks

### PR DESCRIPTION
**Pull request by andrew-moore-octo**
This PR adds support for the new VCS runbooks APIs.

### CODE REVIEW

File: source/Octopus.Client.Tests/Conventions/ClientConventions.cs [hunk 1]
```diff
@@ -74,19 +71,59 @@ public void ThereShouldBeAsyncRepositories()
...
+            var exposedTypes = GetRepoTypesReachableFrom(typeof(IOctopusRepository));
+            
             var missingTypes = SyncRepositoryInterfaceTypes.Except(exposedTypes).ToArray();
             if (missingTypes.Any())
             {
                 Assert.Fail($"All sync *Repository types should be exposed by {nameof(IOctopusRepository)}. Missing: {string.Join(", ", missingTypes.Select(t => t.Name))}");
             }
         }
 
+        private static ISet<Type> GetRepoTypesReachableFrom(Type root)
```
Comment (andrew-moore-octo) [inline source/Octopus.Client.Tests/Conventions/ClientConventions.cs:83]: This is to support the `repo.Beta.Runbooks` chain.

File: source/Octopus.Client.Tests/Conventions/ClientConventions.cs [hunk 2]
```diff
@@ -456,7 +493,7 @@ public void AllAsyncRepositoryInterfacesShouldFollowTheseConventions()
         {
             AsyncRepositoryInterfaceTypes
                 .MustConformTo(Convention.MustLiveInNamespace("Octopus.Client.Repositories.Async"))
-                .AndMustConformTo(Convention.NameMustEndWith("Repository"))
+                .AndMustConformTo(Convention.NameMustEndWith("Repository").Or(Convention.NameMustEndWith("RepositoryBeta")))
```
Comment (TomPeters) [inline source/Octopus.Client.Tests/Conventions/ClientConventions.cs:496]: Should it be called `BetaRepository` instead? Then you don't need this extra convention?

**Review by TomPeters — APPROVED** [review]
Looks good

File: source/Octopus.Client.Tests/Conventions/ClientConventions.cs (hunk 2 above)
Comment (andrew-moore-octo) [reply source/Octopus.Client.Tests/Conventions/ClientConventions.cs]: Good call. I've gone through all the existing `XRepositoryBeta` types and renamed them to `XBetaRepository`, which tidies up this convention test, and means that they'll get covered by all the other convention tests too.

**Comment by andrew-moore-octo** [comment]
[Octopus.Client changes to support VCS runbooks](https://trello.com/c/yVzwTtqW/116-octopusclient-changes-to-support-vcs-runbooks)

**Pull request merged**